models/user.js

```diff
@@ -0,0 +1,43 @@
+import {
+    Schema,
+    model
+} from "mongoose";
+import {
+    ensureFieldUniquity
+} from "./utils";
+
+import bcrypt from "bcrypt";
+import role from './role'
+
+const UserSchema = new Schema({
+    login: {
+        type: String,
+        required: true,
+        unique: true
+    },
+    password: {
+        type: String,
+        required: true
+    },
+    role: {
+        type: [role.schema],
```

Comment:
To get reference to roles of user you can use ref

https://stackoverflow.com/questions/18001478/referencing-another-schema-in-mongoose
[best answer]